### PR TITLE
EWPP-2933: Product and request status notifications.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: src/Notification/Event/Product/StatusChangeAcceptedEvent.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Product;
+
+/**
+ * Event fired when the status of the product changes to "accepted".
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeAcceptedEvent.php:6]:
```suggestion
 * Event fired when the status of the product changes to "Accepted".
```

File: src/Notification/Event/Product/StatusChangeOngoingEvent.php [hunk 2]
```diff
@@ -16,9 +16,9 @@ class StatusChangeOngoingEvent extends BaseEvent
 
     /**
      * @param \OpenEuropa\EPoetry\Notification\Type\Product $product
-     * @param \DateTimeInterface $acceptedDeadline
+     * @param \DateTimeInterface|NULL $acceptedDeadline
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeOngoingEvent.php:19]:
```suggestion
     * @param \DateTimeInterface|null $acceptedDeadline
```

File: src/Notification/Event/Product/StatusChangeAcceptedEvent.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ */
+class StatusChangeAcceptedEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeAcceptedEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Product/StatusChangeCancelledEvent.php [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Product;
+
+/**
+ * Event fired when the status of the product changes to "Cancelled".
+ */
+class StatusChangeCancelledEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeCancelledEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Product/StatusChangeClosedEvent.php [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Product;
+
+/**
+ * Event fired when the status of the product changes to "Closed".
+ */
+class StatusChangeClosedEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeClosedEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Product/StatusChangeReadyToBeSentEvent.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Product;
+
+/**
+ * Event fired when the status of the product changes to "ReadyToBeSent".
+ */
+class StatusChangeReadyToBeSentEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeReadyToBeSentEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Product/StatusChangeSentEvent.php [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Product;
+
+/**
+ * Event fired when the status of the product changes to "Sent".
+ */
+class StatusChangeSentEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeSentEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Product/StatusChangeSuspendedEvent.php [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Product;
+
+/**
+ * Event fired when the status of the product changes to "Suspended".
+ */
+class StatusChangeSuspendedEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Product/StatusChangeSuspendedEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Request/StatusChangeCancelledEvent.php [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Request;
+
+/**
+ * Event fired when the status of the linguistic request changes to "Cancelled".
+ */
+class StatusChangeCancelledEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Request/StatusChangeCancelledEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Request/StatusChangeExecutedEvent.php [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Request;
+
+/**
+ * Event fired when the status of the linguistic request changes to "Executed".
+ */
+class StatusChangeExecutedEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Request/StatusChangeExecutedEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Request/StatusChangeRejectedEvent.php [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Request;
+
+/**
+ * Event ired when the status of the linguistic request changes to "rejected".
+ */
+class StatusChangeRejectedEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Request/StatusChangeRejectedEvent.php:10]:
```suggestion
```

File: src/Notification/Event/Request/StatusChangeSuspendedEvent.php [hunk 12]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace OpenEuropa\EPoetry\Notification\Event\Request;
+
+/**
+ * Event fired when the status of the linguistic request changes to "Suspended".
+ */
+class StatusChangeSuspendedEvent extends BaseEvent
+{
+
```
Comment (sergepavle) [inline src/Notification/Event/Request/StatusChangeSuspendedEvent.php:10]:
```suggestion
```

**Review by sergepavle — APPROVED** [review]

File: src/Notification/NotificationHandler.php [hunk 13]
```diff
@@ -86,30 +94,61 @@ public function receiveNotification(ReceiveNotification $wrapper): ReceiveNotifi
...
             case self::NOTIFICATION_REQUEST_STATUS_CHANGE:
                 $request = $notification->getLinguisticRequest();
-                if ($request->getStatus() === self::REQUEST_STATUS_ACCEPTED) {
-                    $event = new ChangeAcceptedEvent($request, $notification->getPlanningAgent(), $notification->getPlanningSector());
-                }
-                if ($request->getStatus() === self::REQUEST_STATUS_REJECTED) {
-                    $event = new ChangeRejectedEvent($request, $notification->getPlanningAgent(), $notification->getPlanningSector(), $notification->getMessage());
+                switch ($request->getStatus()) {
+                    case self::REQUEST_STATUS_CHANGE_ACCEPTED:
+                        $event = new Request\StatusChangeAcceptedEvent($request, $notification->getPlanningAgent(), $notification->getPlanningSector(), $notification->getMessage());
```
Comment (upchuk) [inline src/Notification/NotificationHandler.php:131]: The request status change accepted *requires* a message? The example doc doesn't have a message. The rejected one has a message.

**Review by ademarco — APPROVED** [review]

**Pull request merged**